### DISCUSSION TITLE
Horizons 4.0

### DISCUSSION THREAD

**Discussion by rfvgyhn** [rfvgyhn/min-ed-launcher#54] [Announcements]
Do note that it is possible for Odyssey owners to launch Horizons 4.0 now, it's possible that it won't be in the future. If FDev decides to implement the same logic the launcher uses in the game client (server side logic), there's nothing we will be able to do to work around it.

# Users that own the Odyssey DLC
1. Update to the [latest] version of min-ed-launcher
2. Update your [settings file] to include the horizons 4.0 product:
    ```json
    "additionalProducts": [{
        "filter": "edh4",
        "directory": "elite-dangerous-odyssey-64",
        "serverargs": "",
        "gameargs": "SeasonTwo",
        "sortkey": "04",
        "product_name": "Elite Dangerous: Horizons (4.0)",
        "product_sku": "FORC-FDEV-DO-38-IN-40"
    }]
    ```
4. Launch your game with the `/edh4` flag

# Users that don't own the Odyssey DLC

I'm not sure if it's the case for all Steam users, but it looks like at least some aren't getting the updated content through Steam (and possibly Epic). If this is the case, you'll see a debug message in your [log file] similar to `2022-09-15 14:43:59.002 -06:00 [DBG] Disabling 'Elite Dangerous: Horizons (4.0)'. Unable to find product at '/mnt/games/Steam/Linux/steamapps/common/Elite Dangerous/Products/FORC-FDEV-DO-38-IN-40/VersionInfo.txt'`

For min-ed-launcher to automatically download the files, you'll need to take the following steps:
1. Update to the [latest] version of min-ed-launcher
2. Navigate to your game's install files
    1. Right click Elite Dangerous in Steam
    2. Manage
    3. Browse local files
3. Create the folder `Products/elite-dangerous-odyssey-64`
4. Copy `VersionInfo.txt` from `Products/elite-dangerous-64` and paste it into `Products/elite-dangerous-odyssey-64`
5. Update your [settings file] to force updates through FDev's servers by adding the 4.0 SKU to `forceUpdate` (comma separated if you already have a SKU set)
    ```
    "forceUpdate": "FORC-FDEV-DO-38-IN-40"
    ```
6. [Optionally] - Horizons 4.0 shares the same filter as Elite Dangerous which will confuse the auto launch flag. Create a new `filterOverrides` entry in your [settings file](https://github.com/rfvgyhn/min-ed-launcher#settings):
    ```
    "filterOverrides": [
        { "sku": "FORC-FDEV-DO-1000", "filter": "edo" },
        { "sku": "FORC-FDEV-DO-38-IN-40", "filter": "edh4" }
    ]
    ```
   Use `/edh4` in your launch options instead of `edh`

[latest]: https://github.com/rfvgyhn/min-ed-launcher/releases/latest
[log file]: https://github.com/rfvgyhn/min-ed-launcher#troubleshooting
[settings file]: https://github.com/rfvgyhn/min-ed-launcher#settings

**Comment by nixRidge** [comment]
You absolutely rock, thank you so much for continuing to develop this handy app :)

**Comment by Johnnii360** [comment]
Hmm... Did all your instructions but get the message that the game can't get access to the auth servers. In the left bottom corner stand `Client access denied`. Btw. this is my first time try to play this game. The game is also linked to Frontier.

Here's my log file: [min-ed-launcher.log](https://github.com/rfvgyhn/min-ed-launcher/files/9913075/min-ed-launcher.log)

Ah and btw. I don't own Odyssee. :)

**Reply by rfvgyhn** [reply]
Does it work with the frontier launcher? This looks like an account linking issue. You only have the single player demo and horizons 4.0 in your purchases response. Normally, you'd also have horizons 3.8, base elite dangerous, and arena.

**Reply by Johnnii360** [reply]
Yep the launcher work now after your how-to and show me also Combat Test and EDH 4.0.

![Bildschirmfoto vom 2022-11-01 20-41-59](https://user-images.githubusercontent.com/15824634/199324578-3aca4ed5-8070-4f07-851a-a3fe075a44b3.png)

**Reply by Johnnii360** [reply]
I also already tried different Proton version but every time got the message that I can't connect to the auth servers and have to check my internet connection. Btw. my Linux is Ubuntu 22.04 LTS (latest).

**Reply by Johnnii360** [reply]
Steam is linked but got this. I don't know if this is necessary?

![Bildschirmfoto vom 2022-11-01 21-23-42](https://user-images.githubusercontent.com/15824634/199333808-03006a05-7f13-49c8-89d0-3f7c77bad173.png)

**Reply by nixRidge** [reply]
That's fine. It's only for those who wish to get Steam keys after buying the game directly from Frontier.

**Reply by rfvgyhn** [reply]
> Yep the launcher work now after your how-to and show me also Combat Test and EDH 4.0.

I'm a little unclear. With the frontier launcher, do you still get the `Client access denied` message or does the game work?

**Reply by Johnnii360** [reply]
> > Yep the launcher work now after your how-to and show me also Combat Test and EDH 4.0.
> 
> I'm a little unclear. With the frontier launcher, do you still get the `Client access denied` message or does the game work?

Still get the message when ED tries to login. Can't access auth servers.

**Reply by rfvgyhn** [reply]
I think you'll have to open a support ticket with frontier. Mentioning the `Client access denied` message and showing them the screenshot of your launcher only having those two entries should get them pointed in the right direction.

**Reply by Johnnii360** [reply]
It works now. I had to redeem the code on the homepage. xD But how to if you didn't get any info about that. It works now. :)

**Comment by DarkStarSword** [comment]
Is it possible to go the other way around and run the Odyssey client connected to the Legacy game? My carrier in Legacy is charging weekly upkeep for a running Bar, which I have no way to shut down in the legacy UI short of decommissioning the entire carrier.

**Reply by rfvgyhn** [reply]
Not that I'm aware of. You could attempt to try different permutations of the `Season*` arguments though. In your log file, there's a json object under `[DBG] Purchases Response` which shows all the products you have access to (run it through a javascript beautifier) which might give you an idea on what things to try.
